components/application-registry/internal/metadata/specification/service.go

```diff
@@ -30,25 +31,25 @@ type Service interface {
 }
 
 type specService struct {
-	minioService minio.Service
-	httpClient   http.Client
+	assetStoreService assetstore.Service
+	downloadClient    download.Client
```

Comment:
Please consider using only one downloadClient (one is inside assetstore Service already). We should either do it now or log a follow-up task